filament/backend/src/vulkan/platform/VulkanPlatformSwapChainImpl.cpp

```diff
@@ -312,6 +322,26 @@ VkResult VulkanPlatformSurfaceSwapChain::recreate() {
     return create();
 }
 
+void VulkanPlatformSurfaceSwapChain::destroy() {
+    if (mSwapChainBundle.depth) {
```

Comment:
I overrided the VulkanPlatformSwapchainImp::destroy since there are semaphores used in the Surface swapchain that need to be destroyed. If they aren't validation errors are raised when the vulkan device is destroyed.

Reply:
Yes, I understand it's necessary.  Can you not just write

```
VulkanPlatformSwapchainImpl::destroy();
```
instead of copying over the code for deleting those images? And then just add to this your code for destroying the semaphore.

Reply:
Ahh ok sorry, didn't understand what you meant in the original comment.